ecs-agent/api/ecs/client/ecs_client.go

```diff
@@ -507,8 +507,10 @@ func (client *ecsClient) SubmitTaskStateChange(change ecs.TaskStateChange) error
 
 func (client *ecsClient) submitTaskStateChange(change ecs.TaskStateChange) error {
 
-	clusterARN := convertTaskToClusterARN(change.TaskARN)
-
+	clusterARN := client.configAccessor.Cluster()
```

Comment:
It seems like the problem here is the cluster arn changing in between tasks, and that does not get updated in the config accessor. Wouldn't this be accessible from the config accessor so long as that were kept up to date? Why do we need an additional field on the TaskStateChange?

Reply:
Firecracker can host multiple tasks on the same instance. So, clusterARN for different tasks will be different since we use task account's clusterARN. We cannot update the configaccessor as we would need to keep updating the member variable every time the API is called for the respective task. It's not that the config accessor is kept up to date in the lifecycle of the task, its just that it would keep differing for different tasks.

Reply:
I see, thanks for explaining!